snippets/ram-console/README.rst

```diff
@@ -0,0 +1,29 @@
+.. _snippet-ram-console:
+
+RAM Console Snippet (ram-console)
+#################################
+
+.. code-block:: console
+
+   west build -S ram-console [...]
+
+Overview
+********
+
+This snippet redirects console output to a RAM buffer, which is linked to
+a dedicated section created by linker scripts through a device tree node
+with compatible string "zephyr,memory-region".
+
+How to add support of a new board
+*********************************
+
+Add board dts overlay to this snippet to add memory-region node with node
+label ``snippet_ram_console`` as the following:
```

Comment:
I am not sure this is how we want folks to use snippets, i.e. a snippet should work "out of the box" without being board specific. In this particular case of course, the presence of the DT node is required but I'm not sure adding an overlay for it *in the snippet* is what we want to recommend?

Reply:
Updated the PR to make it work "out of the box":
Without board specific overlay, it enables the RAM console using a global array for RAM console buffer.